diff -ru aptitude-0.1.4/configure.in aptitude-0.1.4-AX/configure.in
--- aptitude-0.1.4/configure.in	Tue Apr 10 13:25:42 2001
+++ aptitude-0.1.4-AX/configure.in	Wed May 23 00:58:48 2001
@@ -14,8 +14,10 @@
 ac_cv_c_const=yes
 ac_cv_c_inline=yes
 
+LIBS="$LIBS -lrpm -lrpmbuild -lrpmio -lpopt"
+
 dnl Checks for libraries.
-AC_CHECK_LIB(curses, initscr, ,
+AC_CHECK_LIB(ncurses, initscr, ,
 	AC_MSG_ERROR([Can't find the curses libraries -- please install libncurses5-dev]))
 AC_CHECK_LIB(apt-pkg, main, , AC_MSG_ERROR([Can't find the APT libraries -- please install libapt-pkg-dev]))
 AC_CHECK_LIB(pthread, main,
diff -ru aptitude-0.1.4/src/Makefile.am aptitude-0.1.4-AX/src/Makefile.am
--- aptitude-0.1.4/src/Makefile.am	Sun Mar  4 04:34:25 2001
+++ aptitude-0.1.4-AX/src/Makefile.am	Wed May 23 00:58:16 2001
@@ -3,7 +3,7 @@
 SUBDIRS=generic vscreen
 
 localedir = $(datadir)/locale
-INCLUDES = -Wall -Werror -I.. -I$(srcdir) -I$(top_srcdir)/lib -I../intl
+INCLUDES = -Wall -I.. -I$(srcdir) -I$(top_srcdir)/lib -I../intl #-Werror
 DEFS = -DLOCALEDIR=\"$(localedir)\" @DEFS@
 
 bin_PROGRAMS=aptitude
diff -ru aptitude-0.1.4/src/dep_item.cc aptitude-0.1.4-AX/src/dep_item.cc
--- aptitude-0.1.4/src/dep_item.cc	Sun Mar 18 18:53:10 2001
+++ aptitude-0.1.4-AX/src/dep_item.cc	Wed May 23 00:59:42 2001
@@ -30,7 +30,10 @@
 #include "generic/apt.h"
 #include "generic/config_signal.h"
 
-#include <apt-pkg/version.h>
+//#include <apt-pkg/version.h>
+#include <apt-pkg/systemfactory.h>
+
+#define pkgCheckDep _system->checkDep
 
 class pkg_depitem:public pkg_subtree
 {
diff -ru aptitude-0.1.4/src/download.cc aptitude-0.1.4-AX/src/download.cc
--- aptitude-0.1.4/src/download.cc	Sun Apr  8 22:08:43 2001
+++ aptitude-0.1.4-AX/src/download.cc	Wed May 23 00:58:16 2001
@@ -23,6 +23,7 @@
 #include <apt-pkg/error.h>
 #include <apt-pkg/dpkgpm.h>
 #include <apt-pkg/packagemanager.h>
+#include <apt-pkg/systemfactory.h>
 #include <apt-pkg/sourcelist.h>
 #include <apt-pkg/acquire.h>
 #include <apt-pkg/acquire-item.h>
@@ -162,7 +163,9 @@
     }
 
   // Make a package manager, get ready to download
-  pkgDPkgPM pm(*apt_cache_file);
+  //pkgDPkgPM pm(*apt_cache_file);
+  pkgPackageManager *pm_ = _system->CreatePackageManager(*apt_cache_file);
+  pkgPackageManager &pm = *pm_;
   if(!pm.GetArchives(&fetcher, &list, apt_package_records) || _error->PendingError())
     {
       _error->Error(_("Internal error: couldn't generate list of packages to download"));
@@ -237,7 +240,7 @@
 	}
 
       fprintf(f, "Aptitude " VERSION ": log report\n\n");
-      fprintf(f, _("IMPORTANT: this log only lists intended actions; actions which fail due to\ndpkg problems may not be completed.\n\n"));
+      fprintf(f, _("IMPORTANT: this log only lists intended actions; actions which fail due to\nrpm problems may not be completed.\n\n"));
       fprintf(f, _("Will install %li packages, and remove %li packages.\n"), (*apt_cache_file)->InstCount(), (*apt_cache_file)->DelCount());
       if((*apt_cache_file)->UsrSize()>0)
 	fprintf(f, _("%li bytes of disk space will be used\n"), (long int) (*apt_cache_file)->UsrSize());
@@ -313,7 +316,7 @@
       _error->DumpErrors();
       cerr<<_("Ack!  Something bad happened while installing packages.  Trying to recover:")<<endl;
       // and this is really a hack:
-      system("dpkg --configure -a");
+      //system("dpkg --configure -a");
       _error->Discard();
     }
 
Only in aptitude-0.1.4-AX/src: download.cc.orig
diff -ru aptitude-0.1.4/src/generic/Makefile.am aptitude-0.1.4-AX/src/generic/Makefile.am
--- aptitude-0.1.4/src/generic/Makefile.am	Mon Dec  4 04:54:18 2000
+++ aptitude-0.1.4-AX/src/generic/Makefile.am	Wed May 23 00:58:16 2001
@@ -1,7 +1,7 @@
 MAINTAINERCLEANFILES=Makefile.in
 
 localedir = $(datadir)/locale
-INCLUDES = -Wall -Werror -I../../ -I$(srcdir) -I$(top_srcdir)/lib -I../../intl
+INCLUDES = -Wall -I../../ -I$(srcdir) -I$(top_srcdir)/lib -I../../intl #-Werror
 DEFS = -DLOCALEDIR=\"$(localedir)\" @DEFS@
 LDADD = @INTLLIBS@
 
diff -ru aptitude-0.1.4/src/generic/aptcache.cc aptitude-0.1.4-AX/src/generic/aptcache.cc
--- aptitude-0.1.4/src/generic/aptcache.cc	Tue Apr 10 13:09:15 2001
+++ aptitude-0.1.4-AX/src/generic/aptcache.cc	Wed May 23 00:58:16 2001
@@ -33,6 +33,7 @@
 #include <apt-pkg/pkgsystem.h>
 #include <apt-pkg/policy.h>
 #endif
+#include <apt-pkg/systemfactory.h>
 
 #include <vector>
 
@@ -291,7 +292,7 @@
 {
   if(lock==-1)
     return true;
-  string statefile=_config->FindDir("Dir::Aptitude::state", "/var/state/aptitude")+"pkgstates";
+  string statefile=_config->FindDir("Dir::Aptitude::state", "/var/lib/aptitude")+"pkgstates";
 
   FileFd newstate(statefile+".new", FileFd::WriteEmpty);
   if(!newstate.IsOpen())
@@ -694,7 +695,7 @@
 bool aptitudeCacheFile::Open(OpProgress &Progress, bool WithLock)
 {
   if(WithLock)
-    Lock=new pkgDpkgLock;
+    Lock=new pkgRpmLock;	// pkgDpkgLock;
 
   if(_error->PendingError())
     return false;
@@ -705,7 +706,8 @@
 
   if(WithLock)
     {
-      pkgMakeStatusCache(List, Progress);
+      //pkgMakeStatusCache(List, Progress);
+      _system->makeStatusCache(List, Progress);
       if(_error->PendingError())
 	return _error->Error(_("The package lists or status file could not be parsed or opened."));
       if(!_error->empty())
@@ -721,7 +723,8 @@
     }
   else
     {
-      Map=pkgMakeStatusCacheMem(List, Progress);
+      //Map=pkgMakeStatusCacheMem(List, Progress);
+      Map=_system->makeStatusCacheMem(List, Progress);
       Progress.Done();
       if(!Map)
 	return false;
Only in aptitude-0.1.4-AX/src/generic: aptcache.cc.orig
diff -ru aptitude-0.1.4/src/generic/aptcache.h aptitude-0.1.4-AX/src/generic/aptcache.h
--- aptitude-0.1.4/src/generic/aptcache.h	Sun Mar 18 18:53:12 2001
+++ aptitude-0.1.4-AX/src/generic/aptcache.h	Wed May 23 00:58:16 2001
@@ -31,6 +31,7 @@
 #include <config.h>
 
 #include <apt-pkg/depcache.h>
+#include <apt-pkg/rpminit.h>
 
 #include <sigc++/basic_signal.h>
 
@@ -221,7 +222,8 @@
 {
   MMap *Map;
   aptitudeDepCache *Cache;
-  pkgDpkgLock *Lock;
+  //pkgDpkgLock *Lock;
+  pkgRpmLock *Lock;
 public:
   // We look pretty much exactly like a pointer to a dep cache
   inline operator aptitudeDepCache &() {return *Cache;};
Only in aptitude-0.1.4-AX/src/generic: aptcache.h.orig
diff -ru aptitude-0.1.4/src/main.cc aptitude-0.1.4-AX/src/main.cc
--- aptitude-0.1.4/src/main.cc	Fri Mar 23 22:44:16 2001
+++ aptitude-0.1.4-AX/src/main.cc	Wed May 23 00:58:16 2001
@@ -30,7 +30,9 @@
 
 #include <sigc++/func_slot.h>
 
-#include <apt-pkg/init.h>
+//#include <apt-pkg/init.h>
+#include <apt-pkg/rpmfactory.h>
+#include <apt-pkg/rpminit.h>
 
 #include "ui.h"
 
@@ -89,6 +91,11 @@
 
 int main(int argc, char *argv[])
 {
+  {
+    static RPMFactory bla;
+    static pkgRpmLock lock;
+  }
+
   srandom(time(0));
 
   setlocale(LC_ALL, "");
Only in aptitude-0.1.4-AX/src: main.cc.orig
diff -ru aptitude-0.1.4/src/pkg_columnizer.cc aptitude-0.1.4-AX/src/pkg_columnizer.cc
--- aptitude-0.1.4/src/pkg_columnizer.cc	Sun Mar 18 18:53:11 2001
+++ aptitude-0.1.4-AX/src/pkg_columnizer.cc	Wed May 23 00:58:16 2001
@@ -32,6 +32,9 @@
 #ifdef HAVE_LIBAPT_PKG3
 #include <apt-pkg/pkgsystem.h>
 #endif
+#include <apt-pkg/systemfactory.h>
+#define pkgCheckDep _system->checkDep
+
 
 column_definition_list *pkg_item::pkg_columnizer::columns=NULL;
 column_definition_list *pkg_item::pkg_columnizer::status_columns=NULL;
diff -ru aptitude-0.1.4/src/pkg_info_screen.cc aptitude-0.1.4-AX/src/pkg_info_screen.cc
--- aptitude-0.1.4/src/pkg_info_screen.cc	Sat Jan 27 02:12:37 2001
+++ aptitude-0.1.4-AX/src/pkg_info_screen.cc	Wed May 23 00:58:16 2001
@@ -74,11 +74,13 @@
       tree->add_child(new pkg_description(pkg.VersionList(), _("Description")));
       tree->add_child(new vs_staticitem(_("Priority: "),ver.PriorityType()?ver.PriorityType():"Unknown"));
       tree->add_child(new vs_staticitem(_("Section: "),pkg.Section()?pkg.Section():_("Unknown")));
-      tree->add_child(new vs_staticitem(_("Maintainer: "),rec.Maintainer()));
+      //tree->add_child(new vs_staticitem(_("Maintainer: "),rec.Maintainer()));
       tree->add_child(new vs_staticitem(_("Compressed size: "), SizeToStr(ver->Size)));
       tree->add_child(new vs_staticitem(_("Uncompressed size: "), SizeToStr(ver->InstalledSize)));
+#if 0
       if(!rec.SourcePkg().empty())
 	tree->add_child(new vs_staticitem(_("Source Package: "),rec.SourcePkg()));
+#endif
 
       setup_package_deps<pkg_item_with_generic_subtree>(pkg, ver, tree, sig);
     }
Only in aptitude-0.1.4-AX/src: pkg_info_screen.cc.orig
diff -ru aptitude-0.1.4/src/pkg_ver_item.cc aptitude-0.1.4-AX/src/pkg_ver_item.cc
--- aptitude-0.1.4/src/pkg_ver_item.cc	Tue Apr 10 02:21:49 2001
+++ aptitude-0.1.4-AX/src/pkg_ver_item.cc	Wed May 23 00:58:16 2001
@@ -41,6 +41,8 @@
 #include <apt-pkg/configuration.h>
 #include <apt-pkg/strutl.h>
 
+#define pkgCheckDep _system->checkDep
+
 class pkg_ver_columnizer:public column_generator
 {
   pkgCache::VerIterator ver;
Only in aptitude-0.1.4-AX/src: pkg_ver_item.cc.orig
diff -ru aptitude-0.1.4/src/pkg_ver_item.h aptitude-0.1.4-AX/src/pkg_ver_item.h
--- aptitude-0.1.4/src/pkg_ver_item.h	Sun Mar 18 18:53:11 2001
+++ aptitude-0.1.4-AX/src/pkg_ver_item.h	Wed May 23 00:58:16 2001
@@ -30,8 +30,11 @@
 #include "pkg_grouppolicy.h"
 #include "pkg_item_with_subtree.h"
 
-#include <apt-pkg/version.h>
+/* #include <apt-pkg/version.h> */
 #include <apt-pkg/depcache.h>
+#include <apt-pkg/systemfactory.h>
+#include <apt-pkg/depcache.h>
+#define pkgVersionCompare _system->versionCompare           
 #ifdef HAVE_LIBAPT_PKG3
 #include <apt-pkg/pkgsystem.h>
 #endif
Only in aptitude-0.1.4-AX/src: pkg_ver_item.h.orig
diff -ru aptitude-0.1.4/src/vs_progress.cc aptitude-0.1.4-AX/src/vs_progress.cc
--- aptitude-0.1.4/src/vs_progress.cc	Mon Apr  9 22:42:29 2001
+++ aptitude-0.1.4-AX/src/vs_progress.cc	Wed May 23 00:58:16 2001
@@ -31,7 +31,7 @@
 
   if(!Op.empty())
     {
-      assert(Percent>=0 && Percent<=100);
+      //assert(Percent>=0 && Percent<=100);
       barsize=int(Percent*width/100.0);
 
       snprintf(percent_string, 50, ": %i%%", int(Percent));
Only in aptitude-0.1.4-AX/src: vs_progress.cc.orig
