--- aptitude-0.0.7.19.old/configure.in	Fri Mar 23 16:32:50 2001
+++ aptitude-0.0.7.19/configure.in	Fri Mar 23 16:48:53 2001
@@ -17,6 +17,7 @@
 dnl Checks for libraries.
 AC_CHECK_LIB(curses, initscr)
 AC_CHECK_LIB(popt, poptGetArg, , AC_MSG_ERROR([Can't find the popt library -- please install popt-devel]))
+LDFLAGS=-lrpmio
 AC_CHECK_LIB(rpm, rpmdbOpen, , AC_MSG_ERROR([Can't find the RPM library -- please install rpm-devel]))
 AC_CHECK_LIB(apt-pkg, main, , AC_MSG_ERROR([Can't find the APT libraries -- please install libapt-pkg-devel]))
 
@@ -28,7 +29,7 @@
 AC_HEADER_STDC
 AC_CHECK_HEADERS(fcntl.h sys/ioctl.h unistd.h, , AC_MSG_ERROR([Can't find needed header file.]))
 AC_CHECK_HEADER(apt-pkg/init.h, , AC_MSG_ERROR([Can't find the APT header files -- please install libapt-pkg-dev]))
-AC_CHECK_HEADER(rpm/dbindex.h, CXXFLAGS="$CXXFLAGS -I/usr/include/rpm" , AC_MSG_ERROR([Can't find the RPM header files -- please install librpm-devel]))
+AC_CHECK_HEADER(rpm/rpmmacro.h, CXXFLAGS="$CXXFLAGS -I/usr/include/rpm" , AC_MSG_ERROR([Can't find the RPM header files -- please install librpm-devel]))
 AC_CHECK_HEADER(apt-pkg/policy.h, AC_DEFINE(HAVE_LIBAPT_PKG3))
 if test x$HAVE_LIBPTHREAD = x1
 then
